cpp/daal/include/data_management/data/internal/numeric_table_sycl_csr.h

```diff
@@ -0,0 +1,673 @@
+            }
+        }
+        return services::Status();
+    }
+
+public:
+    virtual size_t getDataSize() DAAL_C11_OVERRIDE
+    {
+        if (isCpuTable())
+        {
+            return _cpuTable->getDataSize();
+        }
+
+        return _dataSize;
+    }
+
+protected:
+    template <typename DataType>
+    SyclCSRNumericTable(const services::internal::Buffer<DataType> & bufferData, const services::internal::Buffer<size_t> & bufferColIndices,
+                        const services::internal::Buffer<size_t> & bufferRowOffsets, size_t nColumns, size_t nRows, CSRIndexing indexing,
+                        services::Status & st)
+        : SyclNumericTable(nColumns, nRows, DictionaryIface::equal, st), _indexing(indexing)
+    {
+        _layout   = csrArray;
+        _dataSize = bufferData.size();
+        _defaultFeature.setType<DataType>();
+        st |= _ddict->setAllFeatures(_defaultFeature);
+
+        DAAL_ASSERT(bufferData.size() == bufferColIndices.size());
+        DAAL_ASSERT(bufferRowOffsets.size() == nRows + 1 || !_dataSize);
+
+        if (isCpuContext())
+        {
+            if (!bufferData.size() && !bufferColIndices.size() && !bufferRowOffsets.size())
+            {
+                _cpuTable = CSRNumericTable::create<DataType>(NULL, NULL, NULL, nColumns, nRows, indexing, &st);
+            }
+            else
+            {
+                const services::SharedPtr<DataType> hostData     = bufferData.toHost(ReadWriteMode::readOnly, st);
+                const services::SharedPtr<size_t> hostColIndices = bufferColIndices.toHost(ReadWriteMode::readOnly, st);
+                const services::SharedPtr<size_t> hostRowOffsets = bufferRowOffsets.toHost(ReadWriteMode::readOnly, st);
+                _cpuTable = CSRNumericTable::create(hostData, hostColIndices, hostRowOffsets, nColumns, nRows, indexing, &st);
+            }
+
+            return;
+        }
+        if (_dataSize)
+        {
+            st |= setArrays(bufferData, bufferColIndices, bufferRowOffsets, indexing);
+        }
+    }
+
+    template <typename T>
+    services::Status getTBlock(size_t idx, size_t nrows, int rwFlag, BlockDescriptor<T> & block)
+    {
+        services::throwIfPossible(services::ErrorMethodNotImplemented);
+        return services::ErrorMethodNotImplemented;
+    }
+
+    template <typename T>
+    services::Status releaseTBlock(BlockDescriptor<T> & block)
+    {
+        if (!(block.getRWFlag() & (int)writeOnly)) block.reset();
+        return services::Status();
+    }
+
+    template <typename T>
+    services::Status getTFeature(size_t feat_idx, size_t idx, size_t nrows, int rwFlag, BlockDescriptor<T> & block)
+    {
+        services::throwIfPossible(services::ErrorMethodNotImplemented);
+        return services::ErrorMethodNotImplemented;
+    }
+
+    template <typename T>
+    services::Status releaseTFeature(BlockDescriptor<T> & block)
+    {
+        services::throwIfPossible(services::ErrorMethodNotImplemented);
```

Comment:
Why `releaseTBlock` does not return "not implemented" status as this method does? Why they are inconsistent?

Reply:
Yes, Apply